staging/src/k8s.io/pod-security-admission/test/fixtures_apparmor.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/pod-security-admission/api"
+)
+
+func init() {
+	appArmorFixture_1_0 := fixtureGenerator{
+		expectErrorSubstring: "forbidden AppArmor profile",
+		generatePass: func(pod *corev1.Pod) []*corev1.Pod {
+			pod = ensureAnnotation(pod)
+			return []*corev1.Pod{
+				// container with no annotation
+				tweak(pod, func(copy *corev1.Pod) {
+					name := "test"
+					if len(copy.Spec.Containers) > 0 {
+						copy.Spec.Containers[0].Name = name
+					}
+				}),
+
+				// container with runtime/default annotation
+				tweak(pod, func(copy *corev1.Pod) {
+					name := "test"
+					if len(copy.Spec.Containers) > 0 {
+						copy.Spec.Containers[0].Name = name
+					}
+					copy.Annotations[corev1.AppArmorBetaContainerAnnotationKeyPrefix+name] = "runtime/default"
+				}),
+
+				// container with localhost/foo annotation
+				tweak(pod, func(copy *corev1.Pod) {
+					name := "test"
+					if len(copy.Spec.Containers) > 0 {
+						copy.Spec.Containers[0].Name = name
+					}
```

Comment:
rather than setting the name, just use the existing name. You can also skip the length check, so this just becomes:
```suggestion
					name := copy.Spec.Containers[0].Name
```

Reply:
Same for other tests (init containers could technically be empty, but this is just in a test so we can depend on the minimalViablePod having an init container)